### PR TITLE
EventsPage

### LINKED ISSUE #266: Frontend: EventsPage
Ensure `/events` page is designed properly and that it shows the events only in the future.

See how to do authentication in #255.

### PR DESCRIPTION
**Pull request by nuha90**
# Description

events page

Fixes #266

### CODE REVIEW

File: src/client/containers/EventsPage/EventsPage.js [hunk 1]
```diff
@@ -3,36 +3,47 @@ import { AppHeader } from '../../components/Appheader/AppHeader.component';
...
-    getEvents();
-  }, []);
+    fetch('/api/events').then((data) => {
+      setEvents(data);
+    });
+  }, [fetch]);
 
   return (
-    <div className="events-page">
+    <>
```
Comment (tpetrina) [inline src/client/containers/EventsPage/EventsPage.js:21]: You could use the `Layout` component here.

File: src/client/containers/EventsPage/EventsPage.js [hunk 2]
```diff
@@ -3,36 +3,47 @@ import { AppHeader } from '../../components/Appheader/AppHeader.component';
...
+
   useEffect(() => {
-    const getEvents = async () => {
-      const resp = await fetch('/api/events');
-      const eventsResult = await resp.json();
-      setEvents(eventsResult);
-    };
-    getEvents();
-  }, []);
+    fetch('/api/events').then((data) => {
```
Comment (tpetrina) [inline src/client/containers/EventsPage/EventsPage.js:15]: To make things nicer, add `<Loader` while loading.

File: src/client/containers/EventsPage/EventsPage.styles.css [hunk 3]
```diff
@@ -1,18 +1,20 @@
 .events-page {
-  margin-left: auto;
-  margin-right: auto;
+  margin-left: 30rem;
 }
 
 .group-for-class {
   font-family: Roboto, sans-serif;
   font-weight: bold;
+  left: 50px;
```
Comment (tpetrina) [inline src/client/containers/EventsPage/EventsPage.styles.css:8]: Do not add `left` unless you are doing absolute positioning.

File: src/client/containers/EventsPage/EventsPage.js [hunk 4]
```diff
@@ -3,36 +3,47 @@ import { AppHeader } from '../../components/Appheader/AppHeader.component';
...
-          <CardExampleWithHYF
-            class="events-card"
-            key={event.id}
-            event={event}
-          />
+          <>
+            <div className="group-for-class">
+              Study groups {event.event_type}
+            </div>
+            <div className="events-page">
```
Comment (tpetrina) [inline src/client/containers/EventsPage/EventsPage.js:30]: Don't you want this selector to be in the root element? To style the entire page, not just a single `div`?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Good start.

File: src/client/containers/EventsPage/EventsPage.js [hunk 5]
```diff
@@ -3,36 +3,54 @@ import { AppHeader } from '../../components/Appheader/AppHeader.component';
...
-          <CardExampleWithHYF
-            class="events-card"
-            key={event.id}
-            event={event}
-          />
-        );
-      })}
-      <Footer id="footer" />
-    </div>
+      <Layout className="events-page">
```
Comment (tpetrina) [inline src/client/containers/EventsPage/EventsPage.js:30]: If AppHeader is not inside Layout, how does it look like on the page?

File: src/server/api/controllers/events.controller.js [hunk 6]
```diff
@@ -17,14 +18,16 @@ const createEvent = async (body) => {
...
-      'description',
-    );
+    return await knex('events')
+      .select(
+        'events.id',
+        'events.event_type',
+        'event_date',
+        'venue',
+        'max_participants',
+        'description',
```
Comment (tpetrina) [inline src/server/api/controllers/events.controller.js:28]: If you have time, you could fix these to properly format property names. it should be `eventType`, `eventDate`, etc.

Not necessary, just a nice thing to have.

**Review by tpetrina — APPROVED** [review]

**Review by tpetrina — APPROVED** [review]
🚀

**Review by Pallabi-ghosh — APPROVED** [review]

**Pull request merged**